src/status_im/events.cljs

```diff
@@ -198,7 +198,7 @@
                                      :key-uid (get-in cofx [:db :profile/profile :key-uid])}
                :db                  (assoc db :screens/was-focused-once? true)}
 
-              (= :login view-id)
+              (#{:progress :profiles} view-id)
```

Comment:
feels like we should have tests on events like this to avoid things like this breaking in future. cc @cammellos @ilmotta

Reply:
IMO all non-trivial event handlers should be tested since many of them deal with the business domain. Just as the layer-3 subscriptions that often contain non-trivial logic. Even the re-frame author recommends testing the event layer.

But @J-Son89 this is a lost battle (for me) because I never saw my comments (and code examples) for testing events get traction in `status-mobile`. I'd be very happy to see this situation change in `status-mobile`.

Reply:
could we add a test please @Parveshdhull ?

Reply:
Hi @J-Son89, Thank you very much for your suggestion and @ilmotta for your very valuable input about tests.

Unfortunately, I lack the knowledge required for creating an integration test, and my attempt to write one ended up with a bunch of errors (the login-method for an existing account is not mocked, and after mocking
`Error while saving auth method. The app will continue to work normally, but you will have to log in again next time you launch it.`

So I created an [issue](https://github.com/status-im/status-mobile/issues/16523) for this, So that maybe someone else picks it up, while I get up to speed.

> But @J-Son89 this is a lost battle (for me) 

Maybe, we can discuss this in retro and arrive at consent about tests, and if everyone agrees, maybe we can organize a learn-up section. Thank you

Reply:
I think @ilmotta was referring to a simple event handler test, for example (correct me if I am wrong):

https://github.com/status-im/status-mobile/blob/develop/src/status_im2/contexts/chat/messages/pin/events_test.cljs

I don't think this needs an integration test, though we can have a session about them since it looks like there's some friction.

Reply:
@Parveshdhull, it's a great idea about the retro and discussing this point. I myself never wrote integration tests in `status-mobile`, although I did look at them from time to time. They're valuable, caught some issues in the past as far as I've heard.

Sorry for not being more specific, the tests I was referring to are different/simpler, they are just unit tests for the event handlers, and this is what the re-frame author refers to as well. You basically call the event function defined by `rf/defn` and call that function in the tests passing the inputs and you check the desired data structure is returned (99% of the time a map of effects or nil). Some event handlers using `rf/merge` will be tricky to test since they cause the handler to "absorb" too many responsibilities.

As usual, there are interesting details about testing that only you or the person testing will realize while testing.

I think it's totally understandable if it's not clear how to test them. In case you want to take a look, here are two namespaces with unit tests (not integration tests) for event handlers:

```
- src/status_im2/contexts/chat/events_test.cljs
- src/status_im2/contexts/add_new_contact/events_test.cljs
- src/status_im2/contexts/chat/messages/pin/events_test.cljs
- src/status_im2/contexts/chat/messages/list/events_test.cljs
- src/status_im2/contexts/communities/discover/events_test.cljs
- src/status_im2/contexts/shell/activity_center/events_test.cljs
- src/status_im2/contexts/chat/composer/link_preview/events_test.cljs
- src/status_im2/contexts/chat/messages/delete_message/events_test.cljs
- src/status_im2/contexts/chat/messages/delete_message_for_me/events_test.cljs
```

Reply:
> maybe we can organize a learn-up section.

👍🏼 I can help facilitate this session in early August (unfortunately I'll be off soon in July).

Reply:
> > maybe we can organize a learn-up section.
> 
> 👍🏼 I can help facilitate this session in early August (unfortunately I'll be off soon in July).

I look forward to that, Thank you :heart: